### ISSUE TITLE
`PackerInstall` and other commands from `startup` should all you to pass in a particular plugin name

### ISSUE THREAD

**Issue by tjdevries** [wbthomason/packer.nvim#17]
`:PackerInstall tjdevries/colorbuddy.vim` should just install that. Would be a nice enhancement at some point that I just thought of.

**Comment by wbthomason** [comment]
Do you envision this as modifying your plugin specification file to include some record of you using the plugin? i.e. how does `packer` know that you want to use this plugin persistently across restarts of Neovim? It can detect that it's present by looking in your `packpath`, but without modifying the specification file it has no way to know what plugins in your `packpath` are old and need to be cleaned and which were installed like this.

It's a cool feature idea, though, if we can figure out how this aspect should work!

**Comment by clason** [comment]
Yeah, I had toyed with the idea of key bindings for uninstalling plugins (moving from opt to start or vice versa) from the proposed `PackerList` display, but hit the same brick wall. 

In principle, it could be stored in the compiled `plugins.vim`? (Which then would have to be parsed by `PackerList` as well, making things... thorny.)

**Comment by Shougo** [comment]
The compiled file will be overwritten by `:PackerCompile`.
It should be separated file.

**Comment by wbthomason** [comment]
Right; as @Shougo notes, (1) there's still no way for the compilation process to determine what plugins are stale or just "command installed", and (2) re-compiling the loaders will overwrite the additions.

This roadblock and the problem I'm having in #7 are making me believe that there ought to be some sort of plugin "manifest": a file (probably Lua) written on `install`, `update`, and `compile` (obv. with some logic to prevent redundant writes) that contains the current set of managed plugins and their configuration.

Although this **cannot** be used to make checks for missing/stale plugins faster (e.g. because a user may have made changes to a plugin manually outside of Neovim, by deleting a plugin, checking out a specific commit, etc.), it does allow the following:

- For this feature, the manifest can be structured such that command-installed plugins can be appended automatically. This does not solve the problem of how to get those plugins into the main specification file (if indeed that is desirable), but at least it gives `packer` a persistent record of what plugins the user wants to have.
- For #7, the manifest can be used to regenerate the lazy-loaders when the generated file is older than the specification file. Admittedly, that doesn't solve the problem of regenerating after configuration changes (as the manifest is only regenerated by update/install), so maybe it's not quite the right solution...

Any thoughts?

**Comment by Shougo** [comment]
dein.vim has direct install feature like that.

```
						*dein#direct_install()*
dein#direct_install({repo}[, {options}])
		Direct install a plugin without editing the configuration.
		The arguments are same with |dein#add()|.
		It installs and sources the plugin.
		Note: The direct installed plugins configuration is saved in
		|dein#get_direct_plugins_path()|.  You can load or edit it. >

		" Try deoplete without the configuration.
		call dein#direct_install('Shougo/deoplete.nvim')
<
```

**Comment by wbthomason** [comment]
If I understand correctly, `dein` makes no attempt to add the direct-installed plugins to the user's plugin specification? That may indeed be the way to go.

re: functions vs. commands: I agree - `packer` is structured around functions rather than commands, with a convenience function `packer.startup` that defines some common commands for the user if they choose.

**Comment by Shougo** [comment]
> If I understand correctly, dein makes no attempt to add the direct-installed plugins to the user's plugin specification? That may indeed be the way to go.

Yes.  So, user need to check direct installed plugins by `dein#get_direct_plugins_path()`.

> re: functions vs. commands: I agree - packer is structured around functions rather than commands, 

It is better for me :-)
